### PR TITLE
Make `indent` default in `--show-asr` and `--show-ast`

### LINKED ISSUE #1611: Make `--indent` the default in `--show-asr` and `--show-ast`
I almost always use `--indent`, so it should be the default (just like the color is on by default also). Then let's add `--no-indent` to be used in tests.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #1611.

### CODE REVIEW

File: src/bin/lfortran.cpp [hunk 1]
```diff
@@ -635,7 +635,7 @@ int emit_asr(const std::string &infile,
     } else if (compiler_options.json) {
         std::cout << LCompilers::LFortran::pickle_json(*asr, lm, with_intrinsic_modules) << std::endl;
     } else {
-        std::cout << LCompilers::LFortran::pickle(*asr, compiler_options.use_colors, compiler_options.indent,
+        std::cout << LCompilers::LFortran::pickle(*asr, compiler_options.use_colors, compiler_options.no_indent,
```
Comment (certik) [inline src/bin/lfortran.cpp:638]: I would keep this variable called `indent`, that way no modifications are needed to LFortran except the driver, and in general I think logically that is simpler. The double negation `no_indent = false` is harder for me to reason about what exactly it means.

**Comment by certik** [comment]
Otherwise it looks great, thanks!

File: src/bin/lfortran.cpp [hunk 2]
```diff
@@ -1649,7 +1649,7 @@ int main(int argc, char *argv[])
         app.add_flag("--with-intrinsic-mods", with_intrinsic_modules, "Show intrinsic modules in ASR");
         app.add_flag("--show-ast-f90", show_ast_f90, "Show Fortran from AST for the given file and exit");
         app.add_flag("--no-color", arg_no_color, "Turn off colored AST/ASR");
-        app.add_flag("--indent", compiler_options.indent, "Indented print ASR/AST");
+        app.add_flag("--no-indent", compiler_options.indent, "Indented print ASR/AST");
```
Comment (Thirumalai-Shaktivel) [inline src/bin/lfortran.cpp:1652]:
```suggestion
        app.add_flag("--no-indent", compiler_options.indent, "Turn off indented printing of AST or ASR");
```
or something with similar meaning.

Comment (Pranavchiku) [reply src/bin/lfortran.cpp:1652]: Ahh yes, correct.

**Review by czgdp1807 — APPROVED** [review]

**Comment by czgdp1807** [comment]
Please resolve the conflicts.

**Comment by certik** [comment]
@Pranavchiku let's rebase this on top of the latest main, and ping me. I'll review & merge.

**Comment by Shaikh-Ubaid** [comment]
@certik This is ready.

**Review by certik — APPROVED** [review]
Thanks!

**Comment by Pranavchiku** [comment]
Thank you @Shaikh-Ubaid for completing this PR! I am free now and will start working as soon as possible.

**Pull request merged**